### ISSUE TITLE
Modal: component overview & examples are conflicting

### ISSUE THREAD

**Issue by fspoettel** [salesforce-ux/design-system#240]
Hey Salesforce-UX,
the modal documentation has conflicting information as of `v2.0`. The examples do not match the expected accessibility standards outlined in the component overview. Here is what I caught:

**Expected markup:**

> Modal contains an HTML heading

The example "empty header" shows a modal that does not have an HTML heading.

> Modal has an aria-labelledby attribute whose value is the id of the modal’s heading

None of the examples implement this. The `Notifications/Prompt`-Modules is showing a correct example of the expected modal-markup. How is this supposed to get handled when a modal is rendered with an empty header?

**Comment by kaelig** [comment]
Hi @fspoettel, thank you for using the Lightning Design System and bringing this to our attention.

Let me refer this to the right person and we'll get back to you shortly.

**Comment by SiTaggart** [comment]
Hi @fspoettel 

You are quite correct, some of the modal examples do require updated markup to correct issues like this. As you rightly noted, the Notification Prompt has the correct implementation when a heading is missing, and the Empty Heading modal will be updated in a coming release with the same implementation. In absence of a heading, `aria-label` should be used on the `dialog` container as a replacement.

Notification Prompt was actually only just updated by myself to fix this very problem.

Sorry if this has caused you problems, stay tuned for _lots_ of up coming accessibility improvements on a lot of components.
